ha/k8s_setup/ha_setup.py

```diff
@@ -187,11 +188,21 @@ def process(self):
         Process config command.
         """
         try:
-            consul_endpoint = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints[0]')
-            if not consul_endpoint:
-                sys.stderr.write(f'Failed to get consul config. consul_config: {consul_endpoint}. \n')
+            consul_endpoints = Conf.get(self._index, f'cortx{_DELIM}external{_DELIM}consul{_DELIM}endpoints')
+            #========================================================#
+            # consul Service endpoints from cluster.conf             #
+            #____________________ cluster.conf ______________________#
+            # endpoints:                                             #
+            # - tcp://consul-server.default.svc.cluster.local:8301   #
+            # - http://consul-server.default.svc.cluster.local:8500  #
+            #========================================================#
+            # search for supported consul endpoint url from list of configured consul endpoints
+            filtered_consul_endpoints = list(filter(lambda x: urlparse(x).scheme == const.consul_scheme, consul_endpoints))
+            if not filtered_consul_endpoints or len(filtered_consul_endpoints[0]) == 0:
```

Comment:
As discussed, please check if len(filtered_consul_endpoints[0]) == 0 is required or not

Reply:
done, resolved